syntax = "proto2";
option optimize_for = LITE_RUNTIME;
package webrtc.protobuf_plot;

enum PlotStyle {
  UNDEFINED = 0;
  LINE_GRAPH = 1;
  LINE_DOT_GRAPH = 2;
  BAR_GRAPH = 3;
};

message DataSet {
  repeated float xvalues = 1;
  repeated float yvalues = 2;
  optional string label = 3;
  optional PlotStyle style = 4;
}

message Plot {
  repeated DataSet data_sets = 1;
  optional float xaxis_min = 2;
  optional float xaxis_max = 3;
  optional string xaxis_label = 4;
  optional float yaxis_min = 5;
  optional float yaxis_max = 6;
  optional string yaxis_label = 7;
  optional string title = 8;
}

message PlotCollection {
  repeated Plot plots = 1;
}